src/SFA.DAS.FindApprenticeship.Jobs/Endpoints/HandleVacancyUpdatedEventHttp.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.FindApprenticeship.Jobs.Domain.Handlers;
+using SFA.DAS.FindApprenticeship.Jobs.Infrastructure.Events;
+using System.Net.Http;
+using System.Text.Json;
+
+namespace SFA.DAS.FindApprenticeship.Jobs.Endpoints
+{
+    public class HandleVacancyUpdatedEventHttp
+    {
+        private readonly IVacancyUpdatedHandler _vacancyUpdatedHandler;
+
+        public HandleVacancyUpdatedEventHttp(IVacancyUpdatedHandler vacancyUpdatedHandler)
+        {
+            _vacancyUpdatedHandler = vacancyUpdatedHandler;
+        }
+
+        [FunctionName("HandleVacancyUpdatedEventHttp")]
+        public async Task Run([HttpTrigger(AuthorizationLevel.Function, "get", "post")] HttpRequestMessage req, ILogger log)
+        {
+            log.LogInformation($"HandleVacancyUpdatedEvent HTTP trigger function executed at {DateTime.UtcNow}");
+            var message = JsonSerializer.Deserialize<VacancyUpdatedEvent>(req.Content.ReadAsStream());
```

Comment:
This http trigger is only here for testing really, so it might be overkill to be too defensive, but I wonder if we need to do more in the event of no request body. The request object would exist, but the `.Content.ReadAsStream()` might be problematic (empty stream? serialization error? I wonder what error we get, and whether it's clear to the user what they've done wrong?)

Incidentally, there is an async version of ReadAsStream() and also ReadAsString() which you might also want to consider using (this would probably be fine, as we know the size of the payload is small).

Reply:
I have added some null checks to handle this in the 981 branch now